### DISCUSSION TITLE
[Question] When is there a beta release of 3.0?

### DISCUSSION THREAD

**Discussion by mg-1999** [josephdadams/TallyArbiter#277] [General]
Is it a good idea to release a beta of the current version? 

Would be good to get some user feedback and bugreports on it.

**Comment by JTF4** [comment]
I think we're at a point that we have squashed most of the bugs and can safely release a beta. 

What do you think @josephdadams?

**Comment by josephdadams** [comment]
Sure, go for it

**Comment by MatteoGheza** [comment]
The only problem now is "updating listener clients to the new version".
I'm a bit busy in these days, so I think that I'll work on #272 the next week or so on.
Could someone start updating the documentation?
Documentation TODOs:
- [x] Fix listener clients READMEs to use the same format (see https://github.com/josephdadams/TallyArbiter/commit/eb283dfd29123bf13ffe39e10349be276c58c67a)
- [ ] Add a page in docs explaining breaking changes in v3.0.0
- [x] Update listener clients URLs in docs and add links to READMEs

I think that we can release a beta despite this.

**Reply by hrueger** [reply]
> I think that we can release a beta despite this.

I think so, too, however, I'm currently busy as well and can't do that.

**Reply by JTF4** [reply]
@MatteoGheza @hrueger @josephdadams I updated all the documentation to the best of my ability in #278. What do you think?

**Reply by hrueger** [reply]
LGTM!

I only have one last point I'd like to have done before we merge #278: once we merge it, the new documentation for v3.0.0 beta is live. However, some people might still like to access the old docs or they might get confused if the new listener clients don't work with the current stable version v2.

So it might be a good idea to use Docusaurus' versioning feature: https://docusaurus.io/docs/versioning

Maybe we should just copy the first docs we did with docusaurus from the commit history to the `v2.0.0` version and the new one with all the latest changes to `v3.0.0-beta`.

What do you think?

**Reply by JTF4** [reply]
I'm a fan of that

**Reply by josephdadams** [reply]
@hrueger bold of you to assume people read the documentation, you should check out my inbox sometime 😆

**Reply by hrueger** [reply]
Oops, I see. So you mean it's unnecessary to have the old version still be accessable? Would be interersting to know how many people actually use the docs, sad that GitHub Pages does not have statistics.

**Reply by JTF4** [reply]
I may be able to add an action that injects a google analytics code into the GH-pages branch.

**Reply by josephdadams** [reply]
Personally once we release any version of 3.0 builds, I want people to start using it and I don't want to support 2.x any longer. So we can call it beta if we want or just release it and fix problems as they come up. It's free software, I don't see a need to wait until it's near perfect before we release it. If someone has a breaking production issue, they can always roll back on their own.

The biggest support problems I face are with the m5 clients... all day long I get emails from people who cannot get their m5's to work, whether due to not reading any documentation, or some other problem

**Reply by hrueger** [reply]
Ok, I agree, then we don't need to version the docs.

@JTF4 it seems like there's already a plugin for that: https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-google-analytics

**Comment by mg-1999** [comment]
So the only thing left Todo is update the docs for v3 and roll out the first 3.0 builds? Looks all good to me! Anything I can assist on?

**Reply by josephdadams** [reply]
https://github.com/josephdadams/TallyArbiter/issues/224

If we could do any more tests to what’s on this list just to rule out simple bugs, that would be good. Otherwise it’s fine with me to just fix them as they come up.

**Comment by MatteoGheza** [comment]
Releasing now a beta...

**Reply by wrldtvlr** [reply]
It looks like we'll need to compile 3.0? i.e. the beta isn't available as a CLI install?

**Reply by josephdadams** [reply]
I saw it get published to npm, you should be able to install it the same as any 2.x version

**Reply by wrldtvlr** [reply]
Excellent! Thank you...

**Reply by wrldtvlr** [reply]
Hmmm... Tried npm updating on a Pi, then removing and reinstalling. After install, when running I get the same error both ways. That is: Error: Cannot find module '../dist/index.js'

Looking around /usr/lib/node_modules/tallyarbiter, there is no dist directory

It's possible there is cruft from a previous install, but I'd like to avoid rebuilding the Pi.

**Reply by MatteoGheza** [reply]
> Hmmm... Tried npm updating on a Pi, then removing and reinstalling. After install, when running I get the same error both ways. That is: Error: Cannot find module '../dist/index.js'
> 
> Looking around /usr/lib/node_modules/tallyarbiter, there is no dist directory
> 
> It's possible there is cruft from a previous install, but I'd like to avoid rebuilding the Pi.

Thanks for the beta testing! It should work now.
I'm releasing v3.0.1

**Reply by MatteoGheza** [reply]
Released now. @wrldtvlr can you test it? https://www.npmjs.com/package/tallyarbiter/v/3.0.1
I tried it and it seems to work.

**Reply by wrldtvlr** [reply]
Works like a charm! Now I need to get the M5 Atom clients updated...

**Comment by wrldtvlr** [comment]
Running from the command line works fine, but I'm having a problem with running as a service. Using the 2.x CLI install instructions. I appear to be able to be able to execute 'pm2 start tallyarbiter --name TallyArbiter'. That is, the resulting process list shows the running entry for TallyArbiter as expected. However, I can't reach the web interface.

**Reply by wrldtvlr** [reply]
OK... Looking at the logs, I'm not sure the CLI install is configured correctly for running as a global service. The log shows that it's looking in the /home/pi/tallyarbiter directory for the executable. Shouldn't it be looking in the npm tree?

0|TallyArbiter  | Error: Cannot find module '/home/pi/tallyarbiter'
0|TallyArbiter  |     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
0|TallyArbiter  |     at Function.Module._load (node:internal/modules/cjs/loader:778:27)
0|TallyArbiter  |     at Object.<anonymous> (/usr/lib/node_modules/pm2/lib/ProcessContainerFork.js:33:23)
0|TallyArbiter  |     at Module._compile (node:internal/modules/cjs/loader:1101:14)
0|TallyArbiter  |     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
0|TallyArbiter  |     at Module.load (node:internal/modules/cjs/loader:981:32)
0|TallyArbiter  |     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
0|TallyArbiter  |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
0|TallyArbiter  |     at node:internal/main/run_main_module:17:47 {
0|TallyArbiter  |   code: 'MODULE_NOT_FOUND',
0|TallyArbiter  |   requireStack: []
0|TallyArbiter  | }

**Reply by hrueger** [reply]
Can you run `tallyarbiter` just like that in a terminal window?
If yes, try `which tallyarbiter`. This should give you the full path to the `tallyarbiter` bin file. Then, try to start `pm2` with that path.

**Reply by wrldtvlr** [reply]
That worked! Maybe some cruft left over from a previous local install.
